source/opt/insert_extract_elim.cpp

```diff
@@ -223,6 +428,7 @@ bool InsertExtractElimPass::EliminateInsertExtract(ir::Function* func) {
       }
     }
   }
+  modified |= EliminateDeadInserts(func);
```

Comment:
Since this is just being tacked on to the end, it would be better as it own transformation.  This will create more flexibility, and make the files and classes smaller and easier to read.

Reply:
Sure. Can we do this in a subsequent commit?

Reply:
Sure it can be in another commit.